service/authorization/authorization.go

```diff
@@ -50,7 +51,7 @@ type Config struct {
 	// OAuth Client secret
 	ClientSecret string `mapstructure:"clientsecret" validate:"required"`
 	// OAuth token endpoint
-	TokenEndpoint string `mapstructure:"tokenendpoint" validate:"required,http_url"`
+	tokenEndpoint string
```

Comment:
I'd rather this be left in to allow someone to use an OIDC endpoint that does not support discovery, which AFAIK is optional.

If you do want to require discovery, you still may want to leave this exposed and mark as `//Deprecated` and leave the previous behavior in place (i.e. only use service discovery if not set already)

Reply:
If we need to support an idp that doesn't expose discovery then that functionality should be built into the authn configuration or how we initialize the sdk. 

This configuration was always temporary until we make an ers call outside of rego. Which we will need to do when we transition to the `MatchSubjectMappings` rpc.

Reply:
I guess? As someone who has struggled with keycloak's OIDC discovery configuration a lot, I do think you are making it difficult to work around the very common scenario where discovery is just wrong